app/models/project_step.rb

```diff
@@ -123,4 +139,31 @@ def scheduled_bg
       self.background_color
     end
   end
+
+  #
+  # Translations helpers
+  #
+
+  # todo: refactor up to translatable.rb
+
+  def update_translations!(translation_params)
+    # deleting the translations that have been removed
+    JSON.parse(translation_params[:deleted_locales]).each { |l|
```

Comment:
Seems like the model shouldn't be parsing JSON. Why isn't this happening at the controller level automatically by Rails?